pkg/topology/kademlia/internal/metrics/metrics.go

```diff
@@ -301,6 +312,39 @@ func (c *Collector) IsUnreachable(addr swarm.Address) bool {
 	return cs.ReachabilityStatus != p2p.ReachabilityStatusPublic
 }
 
+type FilterOp func(*Counters) bool
+
+func Unreachable() FilterOp {
```

Comment:
Not sure why this name has a negative connotation, I'd suggest the opposite with a given param, e.g.: `Reachable(bool)`. The same goes for the `Unhealthy` -> `Healthy(bool)`. That way the filter will be more flexible.